bundles/org.openhab.binding.revogi/src/main/java/org/openhab/binding/revogi/internal/RevogiSmartStripDiscoveryService.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.revogi.internal;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.revogi.internal.api.DiscoveryRawResponseDTO;
+import org.openhab.binding.revogi.internal.api.DiscoveryResponseDTO;
+import org.openhab.binding.revogi.internal.api.RevogiDiscoveryService;
+import org.openhab.binding.revogi.internal.udp.DatagramSocketWrapper;
+import org.openhab.binding.revogi.internal.udp.UdpSenderService;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.config.discovery.DiscoveryService;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * The {@link RevogiSmartStripDiscoveryService} helps to discover new smart strips
+ *
+ * @author Andi Bräu - Initial contribution
+ */
+@Component(service = DiscoveryService.class, immediate = true, configurationPid = "discovery.revogismartstripcontrol")
+@NonNullByDefault
+public class RevogiSmartStripDiscoveryService extends AbstractDiscoveryService {
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = new HashSet<>(
+            Collections.singleton(RevogiSmartStripControlBindingConstants.SMART_STRIP_THING_TYPE));
```

Comment:
```suggestion
    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(RevogiSmartStripControlBindingConstants.SMART_STRIP_THING_TYPE);
```

Reply:
that one is done by mvn spotless:apply. The build would fail if I do this change.

Reply:
The suggested change is to remove "new HashSet<>(".

Reply:
ah, I see